plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java

```diff
@@ -32,10 +32,13 @@
 @Log4j2
 public class MLModelCacheHelper {
     private final Map<String, MLModelCache> modelCaches;
+
+    private final Set<String> localDeployedModels;
```

Comment:
How about use HashMap to avoid `synchronized`, https://github.com/ylwu-amzn/ml-commons/blob/2x_autodeploy/plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java#L35

Reply:
Will refresh in the next commit.